Populating the shadow ParticipantProfile table with data from legacy profiles

BroiSatse opened the pull request:
Second step of migrating legacy MentorProfile and EarlyCareerTeacherProfile into STI ParticipantProfile model.

Previous step created the model and made it shadow the legacy profile records via after_save hook. This PR, once merged, will populate the ParticipantProfile table with the existing data in legacy profiles.

Once data is fully migrated we can then refactor the code to use new model in place of legacy profiles.

kicferk1 reviewed (approved):
Who knows how long it will take, but we can give it a go :)

kicferk1 commented:
So what about the users that are added after this migration was run, but before we push any other changes? How are we planning to address those?

BroiSatse commented:
@kicferk1 - This has been already addressed. Previous PR ensured that any change to MentorProfile or EarlyCareerTeacherProfile will be propagated to ParticipantProfile with the same id.

BroiSatse commented:
For completeness - it was supposed to be addressed. It was not.

BroiSatse commented:
Right, so it has been addressed now and we can carry on with this PR. I have removed the migration and moved the code into a temporary rake task instead as per team discussion on slack. Foreign keys will be added in later migrations once the task is run in all environments

kicferk1 commented on app/models/early_career_teacher_profile.rb:
```diff
@@ -15,7 +15,7 @@ class EarlyCareerTeacherProfile < ApplicationRecord
   scope :uplift, -> { sparsity.or(pupil_premium) }
 
-  after_save :save_participant_profile
+  # after_save :save_participant_profile
```
Is this commented out on purpose? Ditto for the mentor profile

BroiSatse replied:
Oof, definitely not - commented it out just to test the task...

kicferk1 commented on lib/tasks/data.rake:
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+namespace :data do
+  desc "Populates PartiicpantProfile using data from EarlyCareerTeacherProfile and MentorProfile"
```
`PartiicpantProfile` has too many `i`s ;)

kicferk1 commented on db/schema.rb:
```diff
@@ -630,7 +630,6 @@
   add_foreign_key "participant_declarations", "lead_providers"
   add_foreign_key "participant_profiles", "cohorts"
   add_foreign_key "participant_profiles", "core_induction_programmes"
-  add_foreign_key "participant_profiles", "participant_profiles", column: "mentor_profile_id"
```
I don't think schema should change in this PR - we don't have any migrations here.

BroiSatse replied:
It is likely this should not go into schema in previous PR... This FK won't work until we have all the recrods moved over to the new table and will be added in next PR...

kicferk1 replied:
ok, so is this schema the result of running migrations from scratch?

BroiSatse replied:
Yes :)
